src/internet_identity/src/storage/anchor.rs

```diff
@@ -195,6 +196,72 @@ impl Anchor {
         timestamps.sort_unstable();
         timestamps.pop().unwrap_or_default()
     }
+
+    /// Returns information about the domains this anchor was active on since the given timestamp.
+    /// Activity on unknown / other domain will be dropped if there is also activity on an II domain
+    /// for the following reasons:
+    /// * no information is most likely caused by the device having been added before we started
+    ///   collecting domain information
+    /// * combinations of an unknown domain and an II domain shows that the anchor is at least partially
+    ///   active on the II domain (but also does non-standard / unsupported things to their anchor).
+    ///   If we are interested in this user group, we might consider extending this function to give
+    ///   them their own [DomainActivity] value.
+    pub fn domain_activity_since(&self, timestamp: Timestamp) -> DomainActivity {
+        #[derive(Default)]
+        struct Accumulator {
+            other: bool,
+            ic0_app: bool,
+            internet_computer_org: bool,
+        }
+
+        let result = self
+            .devices
+            .iter()
+            // filter devices with no activity
+            .filter(|d| {
+                d.last_usage_timestamp
+                    .map(|t| t >= timestamp)
+                    .unwrap_or(false)
+            })
+            // assign domain activity
+            .fold(Accumulator::default(), |mut acc, device| {
+                let Some(ref origin) = device.origin else {
+                    acc.other = true;
+                    return acc;
+                };
+                match origin.as_str() {
+                    IC0_APP_ORIGIN => acc.ic0_app = true,
+                    INTERNETCOMPUTER_ORG_ORIGIN => acc.internet_computer_org = true,
+                    _ => acc.other = true,
+                };
+                acc
+            });
+
+        // Activity on other domains is discarded if there is also activity on an II domain.
+        // The reason is that II might not have complete information since domain information was
+        // only introduced recently.
+        match (result.ic0_app, result.internet_computer_org, result.other) {
```

Comment:
nit: the order of the fields in `match` is different than the order of the fields in the definition of `Accumulator`, which then needs a second look to make sure that the code is correct.  Maybe use the same order in both places?